### PR TITLE
US8: Check if Board is Solved

### PR DESCRIPTION
**Pull request by jzych**
#47

### CODE REVIEW

File: src/Board.cpp [hunk 1]
```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
     return cluesRows_;
 }
 
+bool Board::isSolved()
+{
+    for(auto line : board_)
```
Comment (LordLukin) [inline src/Board.cpp]: extract this loop to a function named checkIfAnyFieldIsUnknown

File: src/Board.cpp [hunk 2]
```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
...
+        if (std::any_of(line.cbegin(),
+                        line.cend(),
+                        [](const auto elem)
+                        {
+                            return elem == BoardFields::UNKNOWN;
+                        }))
+            return false;
+
+    for(int row = 0; row < sizeRows_; row++)
```
Comment (LordLukin) [inline src/Board.cpp]: extracto to a function named verifyRows

File: src/Board.cpp [hunk 3]
```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
...
+        if(not verifyLine(getRow(row), cluesRows_[row]))
+            return false;
+
+    for(int col = 0; col < sizeCols_; col++)
```
Comment (LordLukin) [inline src/Board.cpp]: etract to a function named verifyCols

File: src/Board.cpp [hunk 4]
```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
...
 void Board::display() const
 {
     std::cout << DisplayBoard::display(*this);
 }
+
+bool verifyLine(BLine line, Line clues)
+{
+    Line result;
+    bool continous = false;
+    unsigned int cnt = 0;
```
Comment (LordLukin) [inline src/Board.cpp]: counter?

File: src/Board.cpp [hunk 5]
```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
...
+
+    for(auto elem : line)
```
Comment (LordLukin) [inline src/Board.cpp]: extract to a function named countContinuousBlackFields

File: src/Board.cpp [hunk 6]
```diff
@@ -64,7 +150,59 @@ Lines Board::getCluesRows() const
...
+            result.emplace_back(cnt);
+            cnt = 0;
+            continous = false;
+        }
+    }
+
+    if(continous == true)
+        result.emplace_back(cnt);
+
+    if (result == clues)
```
Comment (LordLukin) [inline src/Board.cpp]: return result == clues;

File: src/Board.cpp (hunk 1 above)
Comment (jzych) [reply src/Board.cpp]: done

File: src/Board.cpp (hunk 2 above)
Comment (jzych) [reply src/Board.cpp]: done

File: src/Board.cpp (hunk 3 above)
Comment (jzych) [reply src/Board.cpp]: done

File: src/Board.cpp (hunk 4 above)
Comment (jzych) [reply src/Board.cpp]: done

File: src/Board.cpp (hunk 5 above)
Comment (jzych) [reply src/Board.cpp]: is this necessery? it will only by one line shorter and verify function will only contain: 
return clues == countCont...;

Comment (jzych) [reply src/Board.cpp]: done

File: src/Board.cpp (hunk 6 above)
Comment (jzych) [reply src/Board.cpp]: done

File: src/Board.cpp [hunk 7]
```diff
@@ -148,7 +148,78 @@ Lines Board::getCluesRows() const
...
+
 void Board::display() const
 {
     std::cout << DisplayBoard::display(*this);
 }
+
+bool Board::checkIfAnyFieldIsUnknown()
+{
+    for(auto line : board_)
+        if (std::any_of(line.cbegin(),
```
Comment (LordLukin) [inline src/Board.cpp]: What do you think about this formatting
```cpp
        return std::any_of(line.cbegin(), line.cend(), [](const auto & elem)
        {
            return elem == BoardFields::UNKNOWN;
        });
```
and a reference in lamba ;)
BTW it looks that you can return result of any_of, without having if

Comment (jzych) [reply src/Board.cpp]: It still have to iterate through all lines of board_.
Best i can do is this:
```
    return std::all_of(board_.begin(), board_.end(), [](const auto line)
    {
        return std::any_of(line.cbegin(),  line.cend(), [](const auto elem)
        {
            return elem == BoardFields::UNKNOWN;
        });
    });
```

**Pull request merged**